tests/lib/rules/no-restricted-imports.js

```diff
@@ -35,6 +35,174 @@ ruleTester.run("no-restricted-imports", rule, {
         {
             code: "import withGitignores from \"foo/bar\";",
             options: [{ patterns: ["foo/*", "!foo/bar"] }]
+        },
+        {
+            code: "import AllowedObject from \"foo\";",
+            options: [{
+                paths: [{
+                    name: "foo",
+                    importNames: ["DisallowedObject"]
+                }]
+            }]
+        },
+        {
+            code: "import DisallowedObject from \"foo\";",
+            options: [{
+                paths: [{
+                    name: "foo",
+                    importNames: ["DisallowedObject"]
+                }]
+            }]
+        },
+        {
+            code: "import * as DisallowedObject from \"foo\";",
+            options: [{
+                paths: [{
+                    name: "bar",
+                    importNames: ["DisallowedObject"],
+                    message: "Please import 'DisallowedObject' from /bar/ instead."
+                }]
+            }],
+            errors: [{
+                message: "* import is invalid because 'DisallowedObject' from 'foo' is restricted.",
+                type: "ImportDeclaration"
+            }]
+        },
+        {
+            code: "import { AllowedObject } from \"foo\";",
+            options: [{
+                paths: [{
+                    name: "foo",
+                    importNames: ["DisallowedObject"],
+                    message: "Please import 'DisallowedObject' from /bar/ instead."
+                }]
+            }],
+            errors: [{
+                message: "'foo' import is restricted from being used. Please import 'DisallowedObject' from /bar/ instead.",
+                type: "ImportDeclaration"
+            }]
+        },
+        {
+            code: "import { DisallowedObject } from \"foo\";",
+            options: [{
+                paths: [{
+                    name: "bar",
+                    importNames: ["DisallowedObject"],
+                    message: "Please import 'DisallowedObject' from /bar/ instead."
+                }]
+            }],
+            errors: [{
+                message: "'foo' import is restricted from being used. Please import 'DisallowedObject' from /bar/ instead.",
+                type: "ImportDeclaration"
+            }]
+        },
+        {
+            code: "import { AllowedObject as DisallowedObject } from \"foo\";",
```

Comment:
I don't understand this test. Although the imported object is being renamed/aliased as `DisallowedObject`, the actual object being imported is `AllowedObject`. Should this not be a valid case?

I see a few other cases that are similar (where the disallowed importName is not actually being imported, but the rule is still reporting error messages).

Am I missing something fundamental? Is `importNames` actually an allow list rather than a block list?

Reply:
@platinumazure, your understanding is correct.  I have removed the erroneous `errors` object from the `valid` test cases.  Because these were part of the `valid` array of tests, the `errors` object was being ignored (which is why my tests were passing, even though it looked like they shouldn't be).

This should be fixed now.  Thanks!

Let me know if you still see any test cases (or .md examples) that seem wrong.